trajopt/include/trajopt/file_write_callback.hpp

```diff
@@ -0,0 +1,26 @@
+#pragma once
+#include <trajopt/common.hpp>
+#include <trajopt/problem_description.hpp>
+#include <trajopt_utils/eigen_conversions.hpp>
+#include <tesseract_core/basic_kin.h>
+#include <tesseract_core/basic_env.h>
+#include <trajopt/plot_callback.hpp>
+#include <functional>
+#include <set>
+#include <fstream>
+#include <string>
+#include <stdio.h>
+#include <Eigen/Core>
+
+typedef std::shared_ptr<std::ofstream> ofstreamPtr;
+
+using namespace Eigen;
+using namespace util;
+using namespace std;
+namespace trajopt
+{
+
+  Optimizer::Callback WriteCallback(string file_name,TrajOptProbPtr prob, vector<string> joint_names,
```

Comment:
Is the lack of references in this function signature intentional? Why not pass joint names as a `const std::vector<std::string>&`?

Reply:
I'm sometimes lazy about this (and I'm not sure it matters for 64bit machines), but Eigen types require certain memory alignment for their SIMD acceleration to function correctly.

So when writing `std::vector<Eigen::Affine3d>`, you should probably write instead: `std::vector<Eigen::Affine3d, Eigen::aligned_allocator<Eigen::Affine3d>>` instead. There's a package in ROS called  `eigen_stl_containers` that provides some aliases for you. It's probably better to define your own using templated aliases.

Reply:
I cannot remember if the pass by value was purposeful. Can you explain to me what passing Eigen's allocator does and why it is preferable?